src/main/java/ru/yandex/practicum/filmorate/storages/UserDbStorage.java

```diff
@@ -0,0 +1,163 @@
+    private final JdbcTemplate jdbcTemplate;
+
+    @Override
+    public Collection<User> findAll() {
+        final String sqlQuery = "select * from users";
+        return jdbcTemplate.query(sqlQuery, this::createUser);
+    }
+
+    private User createUser(ResultSet resultSet, int rowNum) throws SQLException {
+        final int id = resultSet.getInt("id");
+        final String email = resultSet.getString("email");
+        final String login = resultSet.getString("login");
+        final String name = resultSet.getString("name");
+        final LocalDate birthday = resultSet.getDate("birthday").toLocalDate();
+        User user = new User(id, email, login, name, birthday);
+        return user;
+    }
+
+    @Override
+    public User create(User user) {
+        final String sqlQuery = "insert into users(email,login,name,birthday) values(?,?,?,?) ";
+        KeyHolder generatedId = new GeneratedKeyHolder();
+        jdbcTemplate.update(connection -> {
+            final PreparedStatement stmt = connection.prepareStatement(sqlQuery, new String[]{"id"});
+            stmt.setString(1, user.getEmail());
+            stmt.setString(2, user.getLogin());
+            stmt.setString(3, user.getName());
+            stmt.setDate(4, Date.valueOf(user.getBirthday()));
+            return stmt;
+        }, generatedId);
+        user.setId(generatedId.getKey().intValue());
+        log.info("Пользователь с почтой +" + user.getEmail());
+        return user;
+    }
+
+    @Override
+    public User put(User user) {
+        final String checkQuery = "select * from users where id = ?";
+        SqlRowSet userRowSet = jdbcTemplate.queryForRowSet(checkQuery, user.getId());
+        if (!userRowSet.next()) {
+            throw new UserException("Пользователь не найден");
+        }
+        log.info("email = " + user.getEmail() + " login= " + user.getLogin() + " name = " + user.getName() + " birthday= " + user.getBirthday() + " id = " + user.getId());
+        final String query = "UPDATE users SET EMAIL = ?, LOGIN = ?, NAME = ?, BIRTHDAY = ? where id = ?";
+        jdbcTemplate.update(query,
+                user.getEmail(), user.getLogin(), user.getName(), user.getBirthday(), user.getId());
+        return user;
+    }
+
+    @Override
+    public User getUserById(int id) {
+        String checkExists = "select * from users where id=?";
+        SqlRowSet userRowSet = jdbcTemplate.queryForRowSet(checkExists, id);
+        if (!userRowSet.next()) {
+            throw new UserException("Пользователь не найден");
+        }
+        return jdbcTemplate.queryForObject(checkExists, this::createUser, id);
+    }
+
+    @Override
+    public User deleteUserById(int id) {
+        User user = getUserById(id);
+        final String query = "delete from users where id = ?";
+        jdbcTemplate.update(query, id);
+        return user;
+    }
+
+    @Override
+    public Map<Integer, User> getUsers() {
+        Map<Integer, User> userMap = new HashMap();
```

Comment:
Не хватает параметров для `HashMap`. Из-за того, что они подставляются автоматически их очень легко забыть, но если не параметризовать коллекции - проверка на входной тип, работать не будет, что делает программу менее безопасной
```suggestion
        Map<Integer, User> userMap = new HashMap<>();
```